pkg/operator/ceph/csi/betav1csidriver.go

```diff
@@ -72,6 +72,8 @@ func (d beta1CsiDriver) createCSIDriverInfo(ctx context.Context, clientset kuber
 	// As FSGroupPolicy field is immutable, should be set only during create time.
 	// if the request is to change the FSGroupPolicy, we are deleting the CSIDriver object and creating it.
 	if driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil && *driver.Spec.FSGroupPolicy != *csiDriver.Spec.FSGroupPolicy {
+		d.csiClient = csidrivers
```

Comment:
If we are initializing the client and csidriver objects before the call, do we need the initialization in the reCreateCSIDriverInfo func [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/betav1csidriver.go#L91-L92)?

Reply:
we are just creating the variables in the function you defined above if we don't initialize the client and driver object we will hit the nil pointer issue.

Reply:
So the variable declaration in the reCreateCSIDriverInfo function is just for using shortcuts like `csiDriver.Name` in place of `d.csiDriver.Name`. Declaring new variables in the function looks confusing at first, as it looks like the same variable is being declared 2 times(which is not actually happening). It looks cleaner with the variables :+1: